src/vm/database/marf.rs

```diff
@@ -594,3 +594,59 @@ impl ClarityBackingStore for MemoryBackingStore {
         }
     }
 }
+
+impl NullBackingStore {
+    pub fn new() -> Self {
+        NullBackingStore {}
+    }
+
+    pub fn as_clarity_db<'a>(&'a mut self) -> ClarityDatabase<'a> {
+        ClarityDatabase::new(self, &NULL_HEADER_DB, &NULL_BURN_STATE_DB)
+    }
+
+    pub fn as_analysis_db<'a>(&'a mut self) -> AnalysisDatabase<'a> {
+        AnalysisDatabase::new(self)
+    }
+}
+
+impl ClarityBackingStore for NullBackingStore {
```

Comment:
Should all of these methods just panic?  I can't think of an instance where a cost function will do anything that accesses the backing store.